### ISSUE TITLE
Move to setuptools from poetry build no longer installs binary entrypoint

### ISSUE THREAD

**Issue by Torxed** [DataDog/guarddog#92]
This change: https://github.com/DataDog/guarddog/commit/eb1910260168e3c414ce0a71be236f192645a0ab

More specifically this change: https://github.com/DataDog/guarddog/blob/e918bfd4e7497f9b54be1779381eb0dc52391084/pyproject.toml#L64-L66

Causes `~/.local/bin/guarddog` to not be "installed" as a [script entry point](https://setuptools.pypa.io/en/latest/userguide/quickstart.html#entry-points-and-automatic-script-creation) using `pip`.
You can still run `guarddog` with `python -m guarddog scan archinstall` but it will fail for github runners using the `pip install git+https://github.com/DataDog/guarddog.git` method: https://github.com/Torxed/archoffline/actions/runs/3576043149/jobs/6013361779#step:9:5

I'm not sure what setuptools expect during `pip install` to get this working sadly, more used to poetry and flit.
It doesn't seem that this is helping, but it's what I would expect to work:
```toml
[project.scripts]
guarddog = "guarddog.cli:cli"
```

But it's doubtful that this will ever work with the current build system:
```toml
[tool.poetry.scripts]
guarddog = "guarddog.cli:cli"
```

Saving possible related pep saying this should be possible: https://peps.python.org/pep-0660/

**Comment by zmallen** [comment]
`setuptools` requires `console_scripts` section and `setup.py` https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html

Why was this moved from poetry -> setuptools?

**Comment by Torxed** [comment]
Humm, even with `poetry install` there's complications:
```
/usr/sbin/python: No module named guarddog.__main__; 'guarddog' is a package and cannot be directly executed
```
And using `pip install git+https://github.com/DataDog/guarddog.git` (and all requirements) we'll get:
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3.10/site-packages/guarddog/__main__.py", line [5](https://github.com/Torxed/archoffline/actions/runs/3577317329/jobs/6016171522#step:10:6), in <module>
    import guarddog.cli
  File "/usr/lib/python3.10/site-packages/guarddog/cli.py", line 1[7](https://github.com/Torxed/archoffline/actions/runs/3577317329/jobs/6016171522#step:10:8), in <module>
    analyzer = Analyzer()
  File "/usr/lib/python3.[10](https://github.com/Torxed/archoffline/actions/runs/3577317329/jobs/6016171522#step:10:11)/site-packages/guarddog/analyzer/analyzer.py", line 37, in __init__
    self.sourcecode_ruleset = get_rules(".yml", self.sourcecode_path)
  File "/usr/lib/python3.10/site-packages/guarddog/analyzer/analyzer.py", line 34, in get_rules
    return set(rule.replace(file_extension, "") for rule in os.listdir(path) if rule.endswith(file_extension))
FileNotFoundError: [Errno 2] No such file or directory: '/usr/lib/python3.10/site-packages/guarddog/analyzer/sourcecode'
```

**Comment by christophetd** [comment]
@Torxed Can you confirm that the change in https://github.com/DataDog/guarddog/pull/95 looks better to you? At least it works with a `pip install . && guarddog` in a new venv.

**Comment by Torxed** [comment]
> @Torxed Can you confirm that the change in #95 looks better to you? At least it works with a `pip install . && guarddog` in a new venv.

Confirmed, appears to be working :)

**Comment by christophetd** [comment]
Brilliant! Also released v0.1.6 with this new build configuration which seems to be working fine:

```
mkdir /tmp/guarddog-test-install && cd /tmp/guarddog-test-install
python3 -m venv venv && source venv/bin/activate
pip install guarddog==0.1.6
guarddog scan requests
```

Let us know if you have any further issue!

**Comment by Torxed** [comment]
Will do, thank you for the quick turnaround :) Keep it up!
